node/packages/parser/src/ethereum/helpers.ts

```diff
@@ -0,0 +1,10 @@
+import { EthereumSpecific, Tx } from '@shapeshiftoss/blockbook'
+
+export const txMatchesContract = (tx: Tx, contract: string) => {
+  const receiveAddress = tx.vout[0].addresses?.[0] ?? ''
+  return receiveAddress === contract
+}
+
+export const isEthereumSpecific = (
```

Comment:
I can get behind the `txMatchesContract` though maybe a name like `txInteractsWithContract` or something like that would be a little better. I don't quite understand the `isEthereumSpecific` helper and feels like it complicates things instead of just having a optional check inline `!tx.ethereumSpecific?.data`

Reply:
Updated function name to `txInteractsWithContract` in https://github.com/shapeshift/unchained/pull/238/commits/143f663f3aa13c278ce57bef914db30cc1be0599.

You're right about the Type Guard, that was unnecessarily complicated in this case.